common/util.go

```diff
@@ -276,13 +277,13 @@ func (avg *GeometricRollingAverage) GetGeometricRollingAverage() float64 {
 	return float64(*avg)
 }
 
-// ValidateTimeout panics if the passed timeout is unreasonable
-func ValidateTimeout(t time.Duration) {
+// ValidateTimeout returns an error if the passed timeout is unreasonable
+func ValidateTimeout(t time.Duration) error {
 	if t >= time.Millisecond*100 && t <= time.Minute*5 {
-		return
+		return nil
 	}
 
-	panic(fmt.Sprintf(`Configured timeout is out of range: %v`, t))
+	return errors.New(fmt.Sprintf(`Configured timeout is out of range: %v`, t))
```

Comment:
please also specify the allowed range in the error i.e. timeout must be in the range 5min <= timeout >= 100millis.